packages/transaction-controller/src/TransactionController.ts

```diff
@@ -1096,6 +1096,62 @@ export class TransactionController extends BaseController<
     return this.getTransaction(transactionId) as TransactionMeta;
   }
 
+  /**
+   * Update the gas values of a transaction.
+   *
+   * @param transactionId - The ID of the transaction to update.
+   * @param previousGas - Previous gas values to update.
+   * @param previousGas.gasLimit - Maxmimum number of units of gas to use for this transaction.
+   * @param previousGas.maxFeePerGas - Maximum amount per gas to pay for the transaction, including the priority fee.
+   * @param previousGas.maxPriorityFeePerGas - Maximum amount per gas to give to validator as incentive.
+   * @returns The updated transactionMeta.
+   */
+  updatePreviousGasParams(
+    transactionId: string,
+    {
+      gasLimit,
+      maxFeePerGas,
+      maxPriorityFeePerGas,
+    }: {
+      gasLimit?: string;
+      maxFeePerGas?: string;
+      maxPriorityFeePerGas?: string;
+    },
+  ): TransactionMeta {
+    const transactionMeta = this.getTransaction(transactionId);
+
+    if (!transactionMeta) {
+      throw new Error(
+        `Cannot update transaction as no transaction metadata found`,
+      );
+    }
+
+    validateIfTransactionUnapproved(transactionMeta, 'updatePreviousGasParams');
```

Comment:
Not needed now, but the format of all these pending update methods is so similar, including the validations and `pickBy` and update itself, I wonder if we could remove alot of duplication with a private `updateUnapprovedTransaction`. Perhaps a future refactor task.

Reply:
Exactly Matthew, and those update functions located in the scoped files (e.g `updateGasFees`, `updateGas`, `updateTransactionHistory`) only manipulating the passed transaction metadata, having the same naming convention with public update methods (`updateX`). It's a minor one but would be much better if we have clear separation somehow on their name imo.